### PR TITLE
MOSIP-23106

### PR DESCRIPTION
**Pull request by dhanendra06**

### CODE REVIEW

File: id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java [hunk 1]
```diff
@@ -247,6 +250,22 @@ public Map<AllowedKycDto, Object> prepareSharableAttributes(IdResponseDTO idResp
 						attributesMap.put(key, formattedObject);
 					} else if (attribute.equalsIgnoreCase(CredentialConstants.ENCRYPTIONKEY)) {
 						additionalData.put(key.getAttributeName(), credentialServiceRequestDto.getEncryptionKey());
+					}else if(attribute.equalsIgnoreCase(CredentialConstants.VID)){
+						VidInfoDTO vidInfoDTO=null;
```
Comment (LoganathanSekar7627) [inline id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java:254]: You don't need to assign to null, you can just declare it like `VidInfoDTO vidInfoDTO;`

File: id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java [hunk 2]
```diff
@@ -247,6 +250,22 @@ public Map<AllowedKycDto, Object> prepareSharableAttributes(IdResponseDTO idResp
 						attributesMap.put(key, formattedObject);
 					} else if (attribute.equalsIgnoreCase(CredentialConstants.ENCRYPTIONKEY)) {
 						additionalData.put(key.getAttributeName(), credentialServiceRequestDto.getEncryptionKey());
+					}else if(attribute.equalsIgnoreCase(CredentialConstants.VID)){
+						VidInfoDTO vidInfoDTO=null;
+						if(key.getFormat().equalsIgnoreCase(CredentialConstants.RETRIEVE)) {
+							vidInfoDTO = vidUtil.getVIDData(identity.get("UIN").toString(), key.getSource().get(0).getFilter().get(0).getType());
+							if (vidInfoDTO == null) {
+								VidResponseDTO vidResponseDTO = vidUtil.gnerateVID(identity.get("UIN").toString(), key.getSource().get(0).getFilter().get(0).getType());
```
Comment (LoganathanSekar7627) [inline id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java:258]: Type in the method name: gnerateVID

File: id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java [hunk 3]
```diff
@@ -247,6 +250,22 @@ public Map<AllowedKycDto, Object> prepareSharableAttributes(IdResponseDTO idResp
 						attributesMap.put(key, formattedObject);
 					} else if (attribute.equalsIgnoreCase(CredentialConstants.ENCRYPTIONKEY)) {
 						additionalData.put(key.getAttributeName(), credentialServiceRequestDto.getEncryptionKey());
+					}else if(attribute.equalsIgnoreCase(CredentialConstants.VID)){
+						VidInfoDTO vidInfoDTO=null;
+						if(key.getFormat().equalsIgnoreCase(CredentialConstants.RETRIEVE)) {
+							vidInfoDTO = vidUtil.getVIDData(identity.get("UIN").toString(), key.getSource().get(0).getFilter().get(0).getType());
```
Comment (LoganathanSekar7627) [inline id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java:256]: Handle null filter

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java (hunk 1 above)
Comment (dhanendra06) [reply id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java]: taken care

File: id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java (hunk 2 above)
Comment (dhanendra06) [reply id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java]: taken care

File: id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java (hunk 3 above)
Comment (dhanendra06) [reply id-repository/credential-service/src/main/java/io/mosip/credentialstore/provider/CredentialProvider.java]: taken care

**Review by LoganathanSekar7627 — APPROVED** [review]

**Pull request merged**